/**
 * Storage area of TREZOR
 *
 * @author	Marek Palatinus <slush@satoshilabs.com>
 * @version	1
 */

// Sugar for easier handling in Java
option java_package = "com.satoshilabs.trezor.protobuf";
option java_outer_classname = "TrezorStorage";

import "types.proto";

/**
 * Internal persistent storage of device
 */
message Storage {
	required uint32 version = 1;			// version of storage
	optional HDNodeType node = 2;			// BIP-32 node (mnemonic cannot be used if this is present)
	optional string mnemonic = 3;			// BIP-39 mnemonic (node cannot be used if this is present)
	optional bool passphrase_protection = 4;	// whether to require passphrase to decrypt node or stretch mnemonic
	optional uint32 pin_failed_attempts = 5;	// number of failed PIN attempts
	optional string pin = 6;			// current PIN
	optional string language = 7;			// current language
	optional string label = 8;			// device label
	optional bool imported = 9;			// was storage imported from an external source?
}
